File: src/main/java/net/cjsah/console/command/tree/RootCommandNode.java
package net.cjsah.console.command.tree;

import net.cjsah.console.command.builder.ArgumentBuilder;
import net.cjsah.console.command.StringReader;
import net.cjsah.console.command.context.CommandContextBuilder;
import net.cjsah.console.exceptions.CommandException;

public class RootCommandNode extends CommandNode{
    public RootCommandNode() {
        super(null, null, (source) -> true);
    }

    @Override
    public String getName() {
        return "";
    }

    @Override
    public String getUsageText() {
        return "";
    }

    @Override
    protected boolean isValidInput(String input) {
        return false;
    }

    @Override
    public void parse(StringReader reader, CommandContextBuilder builder) throws CommandException {

    }

    @Override
    public ArgumentBuilder<?> createBuilder() {
        throw new IllegalStateException("Cannot convert root into a builder");
    }

    @Override
    protected String getSortedKey() {
        return "";
    }
}

File: src/main/kotlin/net/cjsah/console/command/tree/RootCommandNode.kt
package net.cjsah.console.command.tree

import net.cjsah.console.Console
import net.cjsah.console.command.StringReader
import net.cjsah.console.command.builder.ArgumentBuilder
import net.cjsah.console.command.context.ContextBuilder
import net.cjsah.console.exceptions.CommandException
import net.cjsah.console.exceptions.Para0CommandException
import net.cjsah.console.plugin.Plugin

class RootCommandNode : CommandNode("", null, {true})  {
    override fun getName() = ""

    override fun getUsageText() = ""

    override fun isValidInput(input: String) = false

    fun addChild(plugin: Plugin, node: CommandNode) {
        node.setPlugin(plugin)
        super.addChild(node)
    }

    @Throws(CommandException::class)
    override fun parse(reader: StringReader, builder: ContextBuilder) {
    }

    override fun createBuilder(): ArgumentBuilder<*> {
        throw IllegalStateException("无法将Root转换为builder")
    }

    override fun getSortedKey() = ""
}